extractor/src/main/java/com/example/extractor/CheckstyleExampleExtractor.java

```diff
@@ -129,26 +123,39 @@ private CheckstyleExampleExtractor() {
     public static void main(final String[] args) throws Exception {
         if (args.length < 1) {
             throw new IllegalArgumentException(
-                    "Usage: <checkstyle repo path> [--input-config <config content> "
-                            + "<output file path>]"
+                    "Usage: <checkstyle repo path> [--input-file <input file path> "
+                            + "<config output path> <projects output path>] "
+                            + "or [--config-file <config file path> "
+                            + "<config output path> <projects output path>]"
             );
         }
 
-        if (args.length == SINGLE_INPUT_FILE_ARG_COUNT
-                && "--input-file".equals(args[INPUT_FILE_FLAG_INDEX])) {
-            // New functionality: process single input file
-            final String inputFilePath = args[INPUT_FILE_PATH_INDEX];
-            final String configOutputPath = args[OUTPUT_FILE_PATH_INDEX];
-            final String projectsOutputPath = args[PROJECT_OUTPUT_PATH_INDEX];
-
-            // Process input file and generate config
-            processInputFile(Paths.get(inputFilePath), Paths.get(configOutputPath));
+        if (args.length >= CONFIG_FILE_ARG_COUNT) {
+            final String option = args[CONFIG_FILE_FLAG_INDEX];
+            if ("--input-file".equals(option)) {
+                // Process input file
+                final Path inputFile = Paths.get(args[INPUT_FILE_PATH_INDEX]);
+                final Path configOutputFile = Paths.get(args[CONFIG_OUTPUT_PATH_INDEX]);
+                final Path projectsOutputFile = Paths.get(args[CONFIG_PROJECTS_PATH_INDEX]);
 
-            // Output default projects list
-            outputDefaultProjectsList(projectsOutputPath);
+                processInputFile(inputFile, configOutputFile);
+                outputDefaultProjectsList(projectsOutputFile.toString());
+            }
+            else if ("--config-file".equals(option)) {
```

Comment:
we do not need any few argument in CLI.
as we come to extraction of Check that is in our hardcoded map/collection of modules that have external files, we just put specific file to result folder. 
path to it defined by variable, variable already in config for ExampleX.java.

During execution of regression testing we just need to  set varable to required value, and download such extra file.